src/infra/cache/cache-agent.js

```diff
@@ -55,7 +55,16 @@ class CacheAgent {
     this.promiseReuser = new PromiseReuser();
   }
 
-  contextMiddleware() { // eslint-disable-line class-methods-use-this
```

Comment:
@sbellity since we're creating two caches through CacheAgent and each instance of CachAgent will have its own cache reference, it seems we need to separate this into `workspaceMiddleware` and `contextMiddleware` (or some other way to distinguish between the two types of caches)